src/sidebar/components/MarkdownEditor.tsx

```diff
@@ -173,6 +181,20 @@ function ToolbarButton({
   );
 }
 
+/**
+ * Get the word right before the caret in a textarea.
+ * A word is anything delimited by spaces or newlines.
+ */
+function getTermBeforeCaret(textarea: HTMLTextAreaElement): string {
```

Comment:
I think this would benefit from being moved to a separate module with its own tests, so the nuances of the behavior can be exercised more easily. This could potentially be implemented even without a `textarea` dependency by making the argument any object that satisfies `{ selectionStart: number; value: string; }`.